Havit.Blazor.Components.Web.Bootstrap/TreeViews/Internal/HxTreeViewItemInternal.razor

```diff
@@ -42,19 +48,28 @@
     }
 </div>
 
-@if (this.IsExpanded.Value && hasChildren)
+@if (hasChildren)
 {
-    foreach (var item in children)
-    {
-        <HxTreeViewItemInternal TItem="TItem"
-                        Item="item"
-                        TitleSelector="this.TitleSelector"
-                        IconSelector="this.IconSelector"
-                        CssClass="@this.CssClass"
-                        CssClassSelector="this.CssClassSelector"
-                        ChildrenSelector="this.ChildrenSelector"
-                        Level="@(this.Level + 1)"
-                        ContentTemplate="this.ContentTemplate"
-                        OnItemSelected="this.OnItemSelected" />
-    }
-}
\ No newline at end of file
+    <HxCollapse
+        @ref="collapseComponent"
+        CollapseDirection="CollapseDirection.Vertical"
+        OnShown="HandleAnimationCompleted"
+        OnHidden="HandleAnimationCompleted"
```

Comment:
I think we have to listen to these events and set `IsExpanded` accordingly. If we set `IsExpanded` only when on our expander-click, we can loose other signals where the collapse gets collapsed/expanded. There might not be such case now, but imagine future development where we might want to use accordion-mode of the `HxCollapse` (with `data-bs-parent;) where opening one node automatically closes another.

See `ShowAsync`, `HideAsync`, `HandleJsShown` and `HandleJsHidden` in `HxCollapse.razor.cs` for inspiration.

Reply:
You're correct, however, this brings forth another issue -> the icon will rotate only once the collapse is expanded, which isn't pleasing to watch. Should I solve this issue now? It would probably require listening to the `show` and `hide` events on the collapse using JS in order to notify the `HxTreeViewItem` when the collapse starts animating.

Reply:
Is there any chance to remove the `expanded` class on `hx-tree-view-item-expander` and to rely just on the `show` class being added by the `HxCollapse`?
It is added at the beginning of the animation. Accordion most likely uses this class to control the "arrow" direction (https://getbootstrap.com/docs/5.1/components/accordion/).

Reply:
At the beginning of the animation, the `collapsing` class is added, and the `show` class is added only once the animation is completed. During the animation, only the `collapsing` class is present, so we have no way to tell what direction the arrow should be facing.

Reply:
How is this handled by Bootstrap accordion arrow? @crdo

Reply:
@hakenr It looks for a `collapsed` class on the accordion button, not the collapse itself as shown below.
![image](https://user-images.githubusercontent.com/62853076/181054526-1a6d40e1-9697-4a70-a94d-5c5edea118b6.png)

I'll solve it this way too by using `HxCollapseToggleButton` or `HxCollapseToggleElement`. A `collapsed` class is added or removed right at the start of the animation, exactly what we need.